packages/fxa-content-server/server/lib/routes/get-frontend.js

```diff
@@ -73,6 +73,7 @@ module.exports = function() {
     'sms/sent',
     'sms/sent/why',
     'sms/why',
+    'subscriptions/products/[\\w_]+',
```

Comment:
Can you add a test to a sample product in both [the server side unit tests](https://github.com/mozilla/fxa/blob/39f7dd570e74aa6815912e50a48b92c85a6a6cb2/packages/fxa-content-server/tests/server/routes.js#L66) and [the functional tests](https://github.com/mozilla/fxa/blob/39f7dd570e74aa6815912e50a48b92c85a6a6cb2/packages/fxa-content-server/tests/functional/pages.js#L57).

Reply:
More on this, if I load up http://127.0.0.1:3030/subscriptions/producs/124124 (just a random plan ID), I am redirected to the payments server and then it's an infinite loop of "Plan not found". I click the X to close, and it refreshes to the same thing.

Reply:
I couldn't add a functional test, or at least in that file, as it seems like all those tests specifically expect to land on a page with an element that matches `#stage header` – but this page lands on the payments server where that doesn't exist.

I opened #1730 as a followup about the error message. It's not specifically related to this PR, it just happens the plan-not-found error message uses `location.reload()` on the close button, which isn't sensible here.

Reply:
Yeah, this is basically a 404 condition - there's nowhere for the user to go if they land on a URL with an unknown product or plan ID.

And there isn't really a place for a sample product here, I don't think: The content server doesn't do any look-ups against the auth server for valid products in this implementation, it just redirects over to the payment server and lets that side handle the rest.

Reply:
Oh, and I don't think there's a lot of validation we can do here - a product ID is an opaque `[a-zA-Z0-9_]` string from Stripe without any particular structure we can assert.

Reply:
Can we at assert that? I'll show you the Amplitude logs sometime, people really do shove all sorts of junk in to try to break us.

Reply:
Maybe at a minimum we should not allow `\\w_`, rather just the hex chars + `_`? Otherwise this seems fine, no additional validation needed, invalid strings should 404.